samples/switchless/README.md

```diff
@@ -0,0 +1,133 @@
+
+## About the EDL
+
+First we need to define the functions we want to call between the host and the enclave. To do this we create a `switchless.edl` file:
+
+```edl
+enclave {
+    trusted {
+        public void enclave_helloworld();
+
+    };
+
+    untrusted {
+        void host_helloworld();
+        void host_helloworld_switchless() transition_using_threads;
+    };
+};
+```
+
+Function `host_helloworld_switchless`'s declaration ends with keyword `transition_using_threads`, indicating it should be called switchlessly at run time. However, this a best-effort directive. OE runtime may still choose to fall back to a tradition OCALL if switchless call resources are unavailable, e.g., the enclave is not configured as switchless-capable, or the host worker threads are busy servicing other switchless OCALLs.
+
+To generate the functions with the marshaling code, the `oeedger8r` tool is called in both the host and enclave directories from their Makefiles. For example:
+
+```bash
+cd host
+oeedger8r ../switchless.edl --untrusted --experimental
+```
+
+`oeedger8r` needs the command line flag `--experimental` to be able to recognize the keyword `transition_using_threads`.
+
+## About the host
+
+The host first defines a structure specifically for configuring switchless calls. In this case, we specify the first field `2` as the number of host worker threads for switchless OCALLs. The 2nd field specifies the number of enclave threads for switchless ECALLs. Since switchless ECALL is not yet implementated, we require the 2nd field to be `0`.
+
+```c
+oe_enclave_config_context_switchless_t config = {2, 0};
+```
+
+The host then puts the structure address and the configuration type in an array of configurations for the enclave to be created. Even though we only have one configuration (for switchless) for the enclave, we'd like the flexibility of adding more than one configurations (with different types) for an enclave in the future.
+
+```c
+oe_enclave_config_t configs[] = {{
+        .config_type = OE_ENCLAVE_CONFIG_CONTEXT_SWITCHLESS,
+        .u.context_switchless_config = &config,
+    }};
+```
+
+To make the configurations created above effective, we need to pass the array `configs` into `oe_create_enclave` in the following way:
+
+```c
+oe_create_switchless_enclave(
+             argv[1],
+             OE_ENCLAVE_TYPE_SGX,
+             flags,
+             configs,
+             OE_COUNTOF(configs),
+             &enclave);
+```
+
+The host then makes an ECALL of `enclave_helloworld` to transition into the enclave. After the ECALL returns, the host terminates the enclave.
+
+As shown in the EDL file, the host exposes two host functions: `host_helloworld` and `host_helloworld_switchless`. The former prints "Hello world from regular OCALL", and the latter prints "Hello world from switchless OCALL".
```

Comment:
This sample demonstrates switchless in a case where switchless is inappropriate (i.e. a call that is not frequent).   If we're doing to demonstrate switchless, it should show a case where you'd actually want switchless, and then explain why this case is using switchless, i.e., why it's better than not using it.

Reply:
Fixed with a loop calling the switchless function and explained the rational.